cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1040 @@
+
+
+def _compute_label_quality_scores(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    *,
+    method: str = "objectlab",
+    threshold: Optional[float] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Internal function to prune extra bounding boxes and compute label quality scores based on passed in method."""
+
+    pred_probs_prepruned = False
+    min_pred_prob = _get_min_pred_prob(predictions)
+
+    if threshold is not None:
+        predictions = _prune_by_threshold(
+            predictions=predictions, threshold=threshold, verbose=verbose
+        )
+        if np.abs(min_pred_prob - threshold) < 0.001 and threshold > 0:
+            pred_probs_prepruned = True  # the provided threshold is the threshold used for pre_pruning the pred_probs during model prediction.
+    else:
+        threshold = min_pred_prob  # assume model was not pre_pruned if no threshold was provided
+
+    if method == "objectlab":
+        scores = _get_subtype_label_quality_scores(
+            labels,
+            predictions,
+            alpha=ALPHA,
+            low_probability_threshold=LOW_PROBABILITY_THRESHOLD,
+            high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+            temperature=TEMPERATURE,
+        )
+
+    return scores
+
+
+def _get_min_pred_prob(
+    predictions: List[np.ndarray],
+) -> float:
+    """Returns min pred_prob out of all predictions."""
+    pred_probs = [1.0]  # avoid calling np.min on empty array.
+    for prediction in predictions:
+        for class_prediction in prediction:
+            pred_probs.extend(list(class_prediction[:, -1]))
+
+    min_pred_prob = np.min(pred_probs)
+    return min_pred_prob
+
+
+def _prune_by_threshold(
+    predictions: List[np.ndarray], threshold: float, verbose: bool = True
+) -> List[np.ndarray]:
+    """Removes predicted bounding boxes from predictions who's pred_prob is below the cuttoff threshold."""
+
+    predictions_copy = copy.deepcopy(predictions)
+    num_ann_to_zero = 0
+    total_ann = 0
+    for idx_predictions, prediction in enumerate(predictions_copy):
+        for idx_class, class_prediction in enumerate(prediction):
+            filtered_class_prediction = class_prediction[class_prediction[:, -1] >= threshold]
+            if len(class_prediction) > 0:
+                total_ann += 1
+                if len(filtered_class_prediction) == 0:
+                    num_ann_to_zero += 1
+
+            predictions_copy[idx_predictions][idx_class] = filtered_class_prediction
+
+    p_ann_pruned = total_ann and num_ann_to_zero / total_ann or 0  # avoid division by zero
+    if p_ann_pruned > MAX_ALLOWED_BOX_PRUNE:
+        warnings.warn(
+            f"Pruning with threshold=={threshold} prunes {p_ann_pruned}% labels. Consider lowering the threshold.",
+            UserWarning,
+        )
+    if verbose:
+        print(f"Pruning {num_ann_to_zero} labels out of {total_ann} using threshold=={threshold}.")
```

Comment:
add one more sentence explaining what implication this has. 

Without that additional context, I don't think this print statement is very useful (in which case we could entirely get rid of verbose arg everywhere since this is the only print AFAICT)